lib/livebook_web/live/session_live/app_settings_component.ex

```diff
@@ -125,9 +131,12 @@ defmodule LivebookWeb.SessionLive.AppSettingsComponent do
           <% end %>
         </div>
         <div class="mt-8 flex space-x-2">
-          <.button type="submit" disabled={not @changeset.valid?}>
-            <span>Save</span>
-          </.button>
+          <%= if @live_action == :app_settings do %>
+            <.button type="submit" disabled={not @changeset.valid?}>Save</.button>
+          <% end %>
+          <%= if @live_action == :app_settings_and_launch do %>
+            <.button phx-click="deploy_app" disabled={not @changeset.valid?}>Launch</.button>
```

Comment:
We are already doing deploy as part of the save action.

```suggestion
            <.button disabled={not @changeset.valid?}>Launch</.button>
```